### ISSUE TITLE
Ensure that the current /go-cam web app can operate through the root domain proxy

### ISSUE THREAD

**Issue by kltm** [geneontology/go-site#965]
Currently, the /go-cam web app at geneontology.cloud does not work well with the proxy; ensure that the current /go-cam web app can operate through the root domain proxy.

Originally noted in https://github.com/geneontology/go-site/issues/959#issuecomment-448809876

If this turns out to be onerous, we can use a temporary redirect in the interim.

**Comment by lpalbou** [comment]
It's possibly be due because the go-cam site is a SPA, hence every redirects should be sent to the index.html ?

**Comment by kltm** [comment]
Now using forwarding until proxy fixes in place.

**Comment by lpalbou** [comment]
@kltm do we close this ticket as it's working for the moment ?

**Comment by kltm** [comment]
This is for removing the .cloud extension by making it work via the proxy. There is another ticket (around here somewhere) which was just putting the forwards in place for now.

**Comment by pgaudet** [comment]
@kltm low priority / not planning to do this for Cambridge

**Comment by lpalbou** [comment]
I assigned @kltm to this issue since it's the geneontology.org proxy he manages. 

Any route /gocam/* must redirect to http://geneontology.cloud/*

It should be easy as we only have 3 routes:
* https://geneontology.org/gocam -> https://geneontology.cloud (done)
* https://geneontology.org/gocam/browse -> https://geneontology.cloud/browse
* https://geneontology.org/gocam/docs -> https://geneontology.cloud/docs

**Comment by kltm** [comment]
@lpalbou Now worries as it is not immediately scheduled, but the breakage is occurring in your app, as we found during the initial attempt. When this becomes a little more immediate, we can retry and better document what does wrong or another fronting proxy can be made to simulate.

**Comment by lpalbou** [comment]
I don't think this is coming from the app, I can deploy it under different domain names and it still works so all links are relative. 

However, it's a SPA, so all routes must target the geneontology.cloud/index.html (https://github.com/geneontology/go-site/issues/965#issuecomment-449079760).

If you have some logs or examples during your initial attempt, we can look into it. Thanks

**Comment by kltm** [comment]
@pgaudet Assuming this is bumped up for https://github.com/geneontology/geneontology.github.io/issues/180 ; not sure why I can't add this to related project--my list seems to be limited. Perhaps you'd have better luck?

**Comment by kltm** [comment]
Picking up from gitter.

Unfortunately, the log files for the test are likely long since rotated away. On the upside, we can start building up again on the test configuration or just work it out in a docker image.
The basic test configuration is here:
https://github.com/geneontology/operations/blob/master/ansible/root-domain-head-up-test.yml , via https://github.com/geneontology/operations/blob/master/ansible/roles/webserver-apache-proxy-full/templates/proxy-80.conf .
This is the main proxy from the apache server to the https github-delivered pages. Currently, the /go-cam path just redirects. What we'd probably need is likely another proxy on the /go-cam path to deliver your application.

**Comment by kltm** [comment]
Ah, found the last branch for this work: https://github.com/geneontology/operations/tree/issue-go-site-965-go-cam-cloud-proxy

**Comment by lpalbou** [comment]
Do you remember the type of error / problem you had ? It would be helpful.

Unless you have a better suggestion, I will setup a basic apache server with my geneontology.xyz to see what's going on.
